### PR TITLE
UpdateJob - do not retry BadRequest or Conflict errors from DSA

### LINKED ISSUE #324: Don't retry when validation error (409) from dor-services-app

### PR DESCRIPTION
**Pull request by ndushay**
## Why was this change made?

We shouldn't automatically retry jobs when they have errors that won't resolve themselves.

Fixes #324

## How was this change tested?

unit tests.

### CODE REVIEW

File: app/jobs/update_job.rb [hunk 1]
```diff
@@ -55,6 +55,17 @@ def perform(model_params:, signed_ids:, background_job_result:, start_workflow:
     end
 
     background_job_result.complete!
+  rescue Dor::Services::Client::BadRequestError => e
+    # report as error and do not retry
+    background_job_result.output = { errors: [title: 'HTTP 400 (Bad Request) from dor-services-app',
```
Comment (jmartin-sul) [inline app/jobs/update_job.rb:60]: might want to do the `background_job_result.output.merge` like in the `rescue StandardError` case, so as to keep the `{ druid: model.externalIdentifier }` from line 42?  or, will this always come from `object_client.update` on line 40, implying that `background_job_result.output` will never be set when this error is hit?

File: app/jobs/update_job.rb [hunk 2]
```diff
@@ -55,6 +55,17 @@ def perform(model_params:, signed_ids:, background_job_result:, start_workflow:
...
+                                              message: e.message] }
+    background_job_result.complete!
+  rescue Dor::Services::Client::ConflictResponse => e
+    # RoundtripValidationError for cocina uses HTTP 409 as status
+    # report as error and do not retry
+    background_job_result.output = { errors: [title: 'HTTP 409 (Conflict) from dor-services-app',
```
Comment (jmartin-sul) [inline app/jobs/update_job.rb:66]: same question here

**Review by jmartin-sul — COMMENTED** [review]
one small worry, but otherwise LGTM

**Comment by ndushay** [comment]
@jmartin-sul - good catch - fixed.

File: app/jobs/update_job.rb (hunk 1 above)
Comment (ndushay) [reply app/jobs/update_job.rb]: good point, fixed.

File: app/jobs/update_job.rb (hunk 2 above)
Comment (ndushay) [reply app/jobs/update_job.rb]: good point, fixed.

**Review by jmartin-sul — APPROVED** [review]

**Comment by jmartin-sul** [comment]
> @jmartin-sul - good catch - fixed.

thanks!  i saw that you requested review from justin, so refraining from merging, but feel free to merge if you were just looking for one review to sign off.

**Pull request merged**